limitador/src/storage/infinispan/sets.rs

```diff
@@ -0,0 +1,114 @@
+// Infinispan does not support sets. This module provides some functions on top
+// of the Infinispan client to be able to work with sets.
+
+use crate::storage::infinispan::dist_lock;
+use crate::storage::infinispan::response::response_to_string;
+use crate::storage::StorageErr;
+use infinispan::errors::InfinispanError;
+use infinispan::{request, Infinispan};
+use serde_json::json;
+use std::collections::HashSet;
+use std::iter::FromIterator;
+
+pub async fn get(
+    infinispan: &Infinispan,
+    cache_name: impl AsRef<str>,
+    set_key: impl AsRef<str>,
+) -> Result<HashSet<String>, InfinispanError> {
+    let get_entry_response = infinispan
+        .run(&request::entries::get(cache_name, &set_key))
+        .await?;
+
+    let set = if get_entry_response.status() == 404 {
+        HashSet::new()
+    } else {
+        // TODO: handle other errors
+        serde_json::from_str(&response_to_string(get_entry_response).await).unwrap()
```

Comment:
I think this can only be caused by a programming error. This fails when we try to fetch an entry from infinispan and its body cannot be parsed has a Set. This probably means that limitador wrote something incorrect in that key.

We could return an `InconsistentData` error or similar in a case like this, and deal with it in an upper layer, but I don't think we can do much about it other than exiting and showing a more detailed error explaining what caused the crash.

Something we should do, but not sure if we should get to that point in this first version. I'd rather give a higher priority in trying the distributed cache types offered by Infinispan at this point than dealing with these edge cases.